src/state/operator/selectors.js

```diff
@@ -46,28 +48,33 @@ const totalAvailable = ( { load, capacity } ) => ( capacity - defaultTo( 0, load
 const RESERVE_MAX_CHATS = 2;
 
 /**
-/* Compare function for sorting operator priority. 
- *
- * The first operator in the resulting list should be the next one
- * to be assigned to a chat. This is used for balancing chat load
- * evenly across operators.
+ * Comparison function for sorting operators by the requestingChat property
  *
  * @param {Operator} a First operator to compare
  * @param {Operator} b Second operator to compare
- * @return {Number} -1 if operator a should come first
- *                  0 if both operators are equal priority
- *                  1 if operator b should come first
+ * @return {Number} -1 if operator a is requesting a chat
+ *                  0 if neither are requesting a chat
+ *                  1 if operator b is requesting a chat
  */
-const compareOperatorPriority = ( a, b ) => {
-	// When comparing two operators, always prioritise the one whose
-	// status is available over the one who is in reserve
-	if ( a.status === STATUS_RESERVE && b.status === STATUS_AVAILABLE ) {
-		return 1;
-	}
-	if ( a.status === STATUS_AVAILABLE && b.status === STATUS_RESERVE ) {
-		return -1;
+const compareByRequestForChat = ( a, b ) => {
+	// using ! here incase one is undefined and the other is false
+	if ( ! a.requestingChat === ! b.requestingChat ) {
```

Comment:
Correct but I made a conscious decision here to use the single `!` since the boolean value doesn't actually matter and this function is executed many times so I'm shaving two unnecessary ops from the stack